### DISCUSSION TITLE
Does Celery re-delivers message to the broker queue irrespective of the tool used ?

### DISCUSSION THREAD

**Discussion by vineetsharma883** [celery/celery#7695] [Q&A]
I am running celery worker on k8s. I checked through the [docs](https://docs.celeryq.dev/en/stable/userguide/tasks.html) and found that `the message will be redelivered` in case of worker failure. 

Wanted to confirm above statement should hold true irrespective of the message broker used - `sqs`, `redis` or `rabbitmq` - and message should get re-delivered to the queue if a worker-pod is lost or goes down in the middle of a execution ? i.e. celery would handle re-delivering the message internally irrespective of the broker used

**Comment by auvipy** [comment]
yes that is right as far I know. this was the intended design
